newswires/app/db/FingerpostWireEntry.scala

```diff
@@ -324,16 +324,39 @@ object FingerpostWireEntry
     })
 
     commonWhereClauses match {
-      case Nil => sqls""
+      case Nil => None
       case wherePart :: Nil =>
-        sqls"WHERE $wherePart"
+        Some(wherePart)
       case whereParts =>
-        sqls"WHERE ${sqls.joinWithOr(whereParts.map(clause => sqls"($clause)"): _*)}"
+        Some(sqls.joinWithOr(whereParts.map(clause => sqls"($clause)"): _*))
+    }
+  }
+
+  private[db] def buildWhereClause(
+      searchParams: SearchParams,
+      savedSearchParamList: List[SearchParams],
+      maybeBeforeId: Option[Int],
+      maybeSinceId: Option[Int]
+  ): SQLSyntax = {
+
+    val maybeSearch =
+      buildWhereClauseParts(List(searchParams), maybeBeforeId, maybeSinceId)
+
+    val maybeSavedSearch =
+      buildWhereClauseParts(savedSearchParamList, maybeBeforeId, maybeSinceId)
+
+    (maybeSearch, maybeSavedSearch) match {
+      case (None, None)               => sqls""
+      case (None, Some(bucketSearch)) => sqls"WHERE $bucketSearch"
+      case (Some(userSearch), None)   => sqls"WHERE $userSearch"
+      case (Some(userSearch), Some(bucketSearch)) =>
+        sqls"WHERE ${sqls.joinWithAnd(List(userSearch, bucketSearch).map(clause => sqls"($clause)"): _*)}"
```

Comment:
```suggestion
        sqls"WHERE ($userSearch) AND ($bucketSearch)"
```
for just two clauses, I'd write it directly - joinWithAnd is useful when it's an unknown length list of items, but it's less clear for reading

Reply:
Much better :)